arch/arm/src/tlsr82/tlsr82_serial.c

```diff
@@ -1558,6 +1558,10 @@ static inline int tlsr82_uart_lowputc(int ch)
 #ifndef CONFIG_TLSR82_SPI_SYSLOG
 int up_putc(int ch)
 {
+  /* To make the syslog be thread safe, add sched_lock()/sched_unlock() */
+
+  sched_lock();
```

Comment:
why not use spinlock_irq_save or enter_critical_section

Reply:
Transmit one char with uart cost about 86us (115200 baudrate, 1bit start, 1bit stop, 8bit data). I hope the interrupt disabled time is small, so the interrupt can respond with short delay.

Reply:
> Transmit one char with uart cost about 86us (115200 baudrate, 1bit start, 1bit stop, 8bit data). I hope the interrupt disabled time is small, so the interrupt can respond with short delay.

ADD: `tlsr82_uart_lowputc()` is very fast when the hardware uart tx fifo is not full, but the fifo size is only 8, so in many cases, the fifo is full, and `tlsr82_uart_lowputc()` must wait about 86us (depends on the baudrate) for the free space.

Reply:
OK

Reply:
Done